apps/server/src/modules/oauth/uc/oauth.uc.spec.ts

```diff
@@ -0,0 +1,177 @@
+import axios, { AxiosResponse } from 'axios';
+import jwtDecode from 'jwt-decode';
+import { ObjectId } from 'bson';
+import { IJWT, OauthUc } from '.';
+import { TokenRequestPayload } from '../controller/dto/token-request-payload';
+import { OauthTokenResponse } from '../controller/dto/oauth-token-response';
+import { TokenRequestParams } from '../controller/dto/token-request-params';
+
+describe('OAuthUc', () => {
+	let service: OauthUc;
+	let userRepo: UserRepo;
+	let systemRepo: SystemRepo;
+	let jwtService: FeathersJwtProvider;
+
+	const defaultAuthCode = '43534543jnj543342jn2';
+	const defaultQuery = { code: defaultAuthCode };
+	const defaultErrorQuery = { error: 'Default Error' };
+	const defaultSystemId = '123456789';
+	const defaultDate = new Date();
+	const token =
+		'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiIxMjM0NTY3ODkwIiwibmFtZSI6IkpvaG4gRG9lIiwiaWF0IjoxNTE2MjM5MDIyLCJ1dWlkIjoxMjN9.BZN-sB8PQTfr-xdA1D2TAXWDLTI6fnBkCz_KfW3mCgk';
+	const defaultOauthConfig: OauthConfig = {
+		client_id: '12345',
+		client_secret: 'mocksecret',
+		token_endpoint: 'http://mock.de/mock/auth/public/mockToken',
+		grant_type: 'authorization_code',
+		token_redirect_uri: 'http://mockhost:3030/api/v3/oauth/testsystemId/token',
+		scope: 'openid uuid',
+		response_type: 'code',
+		auth_endpoint: 'mock_auth_endpoint',
+		auth_redirect_uri: '',
+	};
+	const defaultSystem: System = {
+		type: 'iserv',
+		oauthconfig: defaultOauthConfig,
+		id: '',
+		_id: new ObjectId(),
+		createdAt: defaultDate,
+		updatedAt: defaultDate,
+	};
+	const defaultPayloadData: TokenRequestParams = {
+		code: defaultAuthCode,
+		client_id: '12345',
+		client_secret: 'mocksecret',
+		grant_type: 'authorization_code',
+		redirect_uri: 'http://mockhost:3030/api/v3/oauth/testsystemId/token',
+	};
+	const defaultPayload: TokenRequestPayload = {
+		token_endpoint: 'http://mock.de/mock/auth/public/mockToken',
+		tokenRequestParams: defaultPayloadData,
+	};
+
+	beforeEach(async () => {
+		const module: TestingModule = await Test.createTestingModule({
+			imports: [LoggerModule],
+			providers: [
+				OauthUc,
+				{
+					provide: SystemRepo,
+					useValue: {
+						findById(id: EntityId) {
+							return defaultSystem;
+						},
+					},
+				},
+				{
+					provide: UserRepo,
+					useValue: {},
+				},
+				{
+					provide: FeathersJwtProvider,
+					useValue: {
+						generateJwt(user: User) {
+							return user;
+						},
+						// const jwt = (await this.jwtService.generateJwt(user.id)) as string;
+					},
+				},
+			],
+		}).compile();
+
+		service = await module.resolve<OauthUc>(OauthUc);
+		userRepo = await module.resolve<UserRepo>(UserRepo);
+		systemRepo = await module.resolve<SystemRepo>(SystemRepo);
+		jwtService = await module.resolve<FeathersJwtProvider>(FeathersJwtProvider);
+	});
+
+	it('should be defined', () => {
+		expect(service).toBeDefined();
+	});
+
+	// TODO ALOT ALOT
+	describe('startOauth', () => {
```

Comment:
sort tests by unhandled errors, expected errors and acceptance criteria (success states)

Reply:
Ok, we are doing that, tests are not finished yet.